fastq_filtrator.py

```diff
@@ -0,0 +1,199 @@
+    for key in sequence_dictionary:
+        sequence = sequence_dictionary[key][1]
+        gc_composition = ((sequence.count('G')+sequence.count('C'))/len(sequence))*100
+        gc_values.append(round(gc_composition, 1))
+        sequence_dictionary[key].append(round(gc_composition, 1))
+        length_of_reads.append(len(sequence))
+        sequence_dictionary[key].append(len(sequence))
+        quality = []
+        for i in sequence_dictionary[key][3]:
+            quality.append(ord(i)-33)
+        sequence_dictionary[key].append(numpy.mean(quality))
+
+        quality_values.append(numpy.mean(quality))
+    print('Basic statistics on reads quality:\n')
+    print('* GC composition: min =', min(gc_values), ', max =', max(gc_values), ', average =', numpy.mean(gc_values))
+    print('* Length of reads: min =', min(length_of_reads), ', max =', max(length_of_reads),
+          ', average =', numpy.mean(length_of_reads))
+    print('* Quality of reads: min =', min(quality_values), ', max =', max(quality_values),
+          ', average =', numpy.mean(quality_values), '\n\n')
+    return sequence_dictionary
+
+
+def gc_content_filter(sequence_dictionary, gc_bounds):
+    list_to_be_removed = []
+    if gc_bounds != (0, 100):
+        for key in sequence_dictionary.keys():
+            if sequence_dictionary[key][4] < gc_bounds[0] or sequence_dictionary[key][4] > gc_bounds[1]:
+                list_to_be_removed.append(key)
+        for i in list_to_be_removed:
+            sequence_dictionary.pop(i)
+    return sequence_dictionary, list_to_be_removed
+
+
+def length_filter(sequence_dictionary, length_bounds, list_to_be_removed):
+    if length_bounds != (0, 2**32):
+        list_to_be_removed = []
+        for key in sequence_dictionary.keys():
+            if sequence_dictionary[key][5] < length_bounds[0] or sequence_dictionary[key][5] > length_bounds[1]:
+                list_to_be_removed.append(key)
+        for i in list_to_be_removed:
+            sequence_dictionary.pop(i)
+    return sequence_dictionary, list_to_be_removed
+
+
+def quality_filter(sequence_dictionary, quality_threshold, list_to_be_removed):
+    if quality_threshold != 0:
+        list_to_be_removed = []
+        for key in sequence_dictionary.keys():
+            if sequence_dictionary[key][6] < quality_threshold:
+                list_to_be_removed.append(key)
+        for i in list_to_be_removed:
+            sequence_dictionary.pop(i)
+    return sequence_dictionary, list_to_be_removed
+
+
+def saving(sequence_dictionary, sequence_dictionary_original, output_file_prefix, save_filtered, list_to_be_removed):
+    with open((output_file_prefix + "_filtered_reads.fastq"), 'w') as file1:
+        for key in sequence_dictionary.keys():
+            file1.writelines([sequence_dictionary[key][0], sequence_dictionary[key][1],
+                             sequence_dictionary[key][2], sequence_dictionary[key][3]])
+    file1.close()
+    if save_filtered == 'True':
```

Comment:
Лучше использовать логический тип, если у нас всего 2 значения
```suggestion
    if save_filtered == True:
```